src/IECoreScene/MeshAlgoDistributePoints.cpp

```diff
@@ -184,6 +188,16 @@ struct Generator
 				uvBounds.extendBy( uv1 );
 				uvBounds.extendBy( uv2 );
 
+				const float maxCandidatePoints = 1e9;
+				const float approxCandidatePoints = uvBounds.size().x * uvBounds.size().y * textureDensity;
+				if( ! ( approxCandidatePoints <= maxCandidatePoints ) )
+				{
+					std::string e = boost::str( boost::format(
+						"MeshAlgo::distributePoints : Cannot generate more than %i candidate points per polygon. Trying to generate %i. There are circumstances where the output would be reasonable, but this happens during processing due to a polygon with a large area in 3D space which is extremely thin and diagonal in UV space, in which case you may need to clean your UVs. Alternatively, maybe you really want to put an extraordinary number of points on one polygon - please subdivide it before distributing points to help with performance."
```

Comment:
If the area is small but not thin, then we will only visit wang tiles that overlap the tiny area.  There is still a cost in visiting the upper levels of the hierarchy, but this is basically the same as for any highly subdivided mesh - we're never particularly efficient, but you don't get the really nasty case of thin and diagonal where we end up visiting huge numbers of leaf tiles that don't even overlap the target triangle at all.  If the area is small and thin but not diagonal, then that is moderately bad, though definitely less bad than thin and diagonal - I think basically we're not visiting tiles that don't overlap the triangle at all, but a thin triangle can still span a bunch of leaf tiles.  I'm up for simply omitting "and diagonal", since any cleaning process would probably not be specific to diagonals, but I think "thin" is important to what triggers a dramatic slowdown.

Reply:
Yeah, that alternate PointDistribution call could be interesting - I'm not sure it would be worth it for the thin triangle special case, but if you were generating a very large number of points on a fairly low polygon mesh, you would theoretically save half the points that need testing. Testing the tile bounds against the triangle would have a cost though - maybe you would need a metric where if there are less than <someNumber> points left to be generated, then it's worth testing if we can skip the current tile.